template/src/App.tsx

```diff
@@ -39,58 +54,129 @@ if (Platform.OS === 'ios') {
 }
 
 const App: React.FC = () => {
-  const [phrase, onChangePhrase] = useState('');
-
+  const {join, create} = useFpe((data) =>
+    data?.components ? data.components : {},
+  );
+  const videoCall = useFpe((data) =>
+    typeof data?.components?.videoCall !== 'object'
+      ? isValidElementType(data?.components?.videoCall)
+      : undefined,
+  );
+  const CustomRoutes = useFpe((data) => data?.customRoutes);
+  const AppRoot = useFpe((data) => data?.appRoot);
+  const CreateCmp = cmpTypeGuard(Create, create);
+  const RootWrapper = getCmpTypeGuard(React.Fragment, AppRoot);
   return (
-    <ImageBackground
-      source={{uri: $config.BG}}
-      style={{flex: 1}}
-      resizeMode={'cover'}>
-      <SafeAreaView style={{flex: 1}}>
-        <StatusBar hidden={true} />
-        <Toast ref={(ref) => Toast.setRef(ref)} config={ToastConfig} />
-        <StorageProvider>
-          <GraphQLProvider>
-            <Router>
-              <SessionProvider>
-                <ColorConfigure>
-                  <Navigation />
-                  <Switch>
-                    <Route exact path={'/'}>
-                      <Redirect to={'/create'} />
-                    </Route>
-                    <Route exact path={'/authenticate'}>
-                      {shouldAuthenticate ? <OAuth /> : <Redirect to={'/'} />}
-                    </Route>
-                    <Route path={'/auth-token/:token'}>
-                      <StoreToken />
-                    </Route>
-                    <Route exact path={'/join'}>
-                      <Join phrase={phrase} onChangePhrase={onChangePhrase} />
-                    </Route>
-                    {shouldAuthenticate ? (
-                      <PrivateRoute
-                        path={'/create'}
-                        failureRedirectTo={'/authenticate'}>
-                        <Create />
-                      </PrivateRoute>
-                    ) : (
-                      <Route path={'/create'}>
-                        <Create />
-                      </Route>
-                    )}
-                    <Route path={'/:phrase'}>
-                      <VideoCall />
-                    </Route>
-                  </Switch>
-                </ColorConfigure>
-              </SessionProvider>
-            </Router>
-          </GraphQLProvider>
-        </StorageProvider>
-      </SafeAreaView>
-    </ImageBackground>
+    <RootWrapper>
```

Comment:
Take this out as a separate appConfigure provider components and use it by passing chindren props,
<RootWrapper>
      <ImageBackground
        source={{uri: $config.BG}}
        style={{flex: 1}}
        resizeMode={'cover'}>
        <SafeAreaView style={{flex: 1}}>
          <StatusBar hidden={true} />
          <Toast ref={(ref) => Toast.setRef(ref)} config={ToastConfig} />
          <StorageProvider>
            <GraphQLProvider>
              <Router>
                <SessionProvider>
                  <ColorConfigure>
                    <DimensionProvider>
                      <LanguageProvider>
                        <ErrorProvider>
                          <Error />
                          <Navigation />
                          <Switch>

Reply:
Done